store/sqlstore/user_store.go

```diff
@@ -1055,20 +1045,41 @@ func (us SqlUserStore) PermanentDelete(userId string) store.StoreChannel {
 	})
 }
 
-func (us SqlUserStore) AnalyticsUniqueUserCount(teamId string) store.StoreChannel {
+func (us SqlUserStore) Count(options model.UserCountOptions) store.StoreChannel {
 	return store.Do(func(result *store.StoreResult) {
-		query := ""
-		if len(teamId) > 0 {
-			query = "SELECT COUNT(DISTINCT Users.Email) From Users, TeamMembers WHERE TeamMembers.TeamId = :TeamId AND Users.Id = TeamMembers.UserId AND TeamMembers.DeleteAt = 0 AND Users.DeleteAt = 0"
+		query := sq.Select("COUNT(Users.Id)").From("Users")
+
+		if !options.IncludeDeleted {
+			query = query.Where("Users.DeleteAt = 0")
+		}
+
+		if options.IncludeBotAccounts {
+			if options.ExcludeRegularUsers {
+				query = query.LeftJoin("Bots ON Users.Id = Bots.UserId").Where("Bots.UserId IS NOT NULL")
+			}
 		} else {
-			query = "SELECT COUNT(DISTINCT Email) FROM Users WHERE DeleteAt = 0"
+			query = query.LeftJoin("Bots ON Users.Id = Bots.UserId").Where("Bots.UserId IS NULL")
+			if options.ExcludeRegularUsers {
+				// Currenty this doesn't make sense because it will always return 0
```

Comment:
nit: Seems fine to just return an empty result vs. a special error?

Reply:
I think I prefer to fail here, because if you are asking for no users and no bots, you don't need to ask at all. Maybe is more forgiving to return an empty result, but is less error-prone to fail here forcing the developer to check his own logic while is depeloping it.